medical/models/res_partner.py

```
@@ -1,85 +1,39 @@
-        help='Check if the party is a patient',
-    )
+    type = fields.Selection(selection_add=[
+        ('medical.patient', 'Medical Patient'),
+    ])
     alias = fields.Char(
         string='Alias',
-        size=256,
-        help='Common name that the Party is reffered',
-    )
-    activation_date = fields.Date(
-        string='Activation date',
-        help='Date of activation of the party',
-    )
-    last_name = fields.Char(
-        string='Last Name',
-        size=256,
-        help='Last Name',
-    )
-    is_work = fields.Boolean(string='Work')
-    is_person = fields.Boolean(
-        string='Person',
-        help='Check if the party is a person.',
-    )
-    is_school = fields.Boolean(string='School')
-    is_pharmacy = fields.Boolean(
-        string='Pharmacy',
-        help='Check if the party is a Pharmacy',
-    )
-    is_insurance_company = fields.Boolean(
-        string='Insurance',
-        help='Check if the party is a patient',
-    )
-    ref = fields.Char(
-        size=256,
-        string='ID/SSN',
-        help='Patient Social Security Number or equivalent',
+        help='Common name that the Party is referred',
     )
     patient_ids = fields.One2many(
-        'medical.patient',
-        fields_id='medical_center_id',
         string='Related Patients',
+        comodel_name='medical.patient',
+        inverse_name='partner_id',
     )
+
+    @api.model
+    def create(self, vals):
+        """ It overloads create to bind appropriate medical entity """
+        try:
+            if all((
+                vals['type'].startswith('medical.'),
+                not self.env.context.get('medical_entity_no_create'),
+            )):
+                model = self.env[vals['type']].with_context(
+                    medical_entity_no_create=True,
+                )
+                medical_entity = model.create(vals)
+                return medical_entity.partner_id
+        except (AttributeError, KeyError):
```

I feel worried when reading this :)

I think there should be add a comment here to explain why the exception is ignored voluntarily. Maybe a warning could also be logged.

---

Yeah this is technically breaking Python code by excepting without an output, but I figured it would litter the logs. I will add a comment and I think I can do a little bit for legibility at the sacrifice of a bit of McCabe rating